complete/src/main/java/com/google/example/android/basicactivityrecognitiontransitionsample/MainActivity.java

```diff
@@ -0,0 +1,306 @@
+    private LogFragment mLogFragment;
+
+    private static String toActivityString(int activity) {
+        switch (activity) {
+            case DetectedActivity.STILL:
+                return "STILL";
+            case DetectedActivity.WALKING:
+                return "WALKING";
+            default:
+                return "UNKNOWN";
+        }
+    }
+
+    private static String toTransitionType(int transitionType) {
+        switch (transitionType) {
+            case ActivityTransition.ACTIVITY_TRANSITION_ENTER:
+                return "ENTER";
+            case ActivityTransition.ACTIVITY_TRANSITION_EXIT:
+                return "EXIT";
+            default:
+                return "UNKNOWN";
+        }
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        setContentView(R.layout.activity_main);
+        Toolbar toolbar = findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+        mLogFragment =
+                (LogFragment) getSupportFragmentManager().findFragmentById(R.id.log_fragment);
+
+        activityTrackingEnabled = false;
+
+        // List of activity transitions to track.
+        activityTransitionList = new ArrayList<>();
+
+        // TODO: Add activity transitions to track.
+        activityTransitionList.add(new ActivityTransition.Builder()
+                .setActivityType(DetectedActivity.WALKING)
+                .setActivityTransition(ActivityTransition.ACTIVITY_TRANSITION_ENTER)
+                .build());
+        activityTransitionList.add(new ActivityTransition.Builder()
+                .setActivityType(DetectedActivity.WALKING)
+                .setActivityTransition(ActivityTransition.ACTIVITY_TRANSITION_EXIT)
+                .build());
+        activityTransitionList.add(new ActivityTransition.Builder()
+                .setActivityType(DetectedActivity.STILL)
+                .setActivityTransition(ActivityTransition.ACTIVITY_TRANSITION_ENTER)
+                .build());
+        activityTransitionList.add(new ActivityTransition.Builder()
+                .setActivityType(DetectedActivity.STILL)
+                .setActivityTransition(ActivityTransition.ACTIVITY_TRANSITION_EXIT)
+                .build());
+
+        // TODO: Initialize PendingIntent that will be triggered when a activity transition occurs.
+        Intent intent = new Intent(TRANSITIONS_RECEIVER_ACTION);
+        mActivityTransitionsPendingIntent =
+                PendingIntent.getBroadcast(MainActivity.this, 0, intent, 0);
+
+        // TODO: Create and register a BroadcastReceiver to listen for activity transitions.
+        // The receiver listens for the PendingIntent above that is triggered by the system when an
+        // activity transition occurs.
+        mTransitionsReceiver = new TransitionsReceiver();
+        registerReceiver(mTransitionsReceiver, new IntentFilter(TRANSITIONS_RECEIVER_ACTION));
```

Comment:
Can this be via LocalBroadcastManager to reduce the risk of leaking the BroadcastReceiver?
I'm not familiar with the ActivityRecognition, so I'm just checking in case.

Reply:
Good point, I was using their original code for that part. I will rework it.

Reply:
Fixed.